tools/docker-entrypoint.sh

```diff
@@ -4,18 +4,24 @@
 
 networkdatadir="${LIGHTNINGD_DATA}/${LIGHTNINGD_NETWORK}"
 
-if [ "$EXPOSE_TCP" == "true" ]; then
-    set -m
-    lightningd "$@" &
+set -m
+lightningd "$@" &
 
-    echo "Core-Lightning starting"
-    while read -r i; do if [ "$i" = "lightning-rpc" ]; then break; fi; done \
+echo "Core-Lightning starting"
+while read -r i; do if [ "$i" = "lightning-rpc" ]; then break; fi; done \
     < <(inotifywait -e create,open --format '%f' --quiet "${networkdatadir}" --monitor)
-    echo "Core-Lightning started"
+
+if [ "$EXPOSE_TCP" == "true" ]; then
     echo "Core-Lightning started, RPC available on port $LIGHTNINGD_RPC_PORT"
 
     socat "TCP4-listen:$LIGHTNINGD_RPC_PORT,fork,reuseaddr" "UNIX-CONNECT:${networkdatadir}/lightning-rpc" &
-    fg %-
-else
-    exec lightningd --network="${LIGHTNINGD_NETWORK}" "$@"
```

Comment:
Seems we drop support for not exposing here? That's fine imho, since I don't think anybody wants to `docker exec` into the container in order to interact anyway, and exposing is likely done by the vast majority.

Reply:
I get `Wrong network! Our Bitcoin backend is running on 'regtest', but we expect 'main'.` with  `LIGHTNINGD_NETWORK := regtest` when param `--network` is not provided. 
What is the reason, that it is not passed to the daemon?

Reply:
Ah, yes, the line I moved was missing --network!  Will fix.

To clarify:
1. With EXPOSE_TCP false, this script just did `exec lightningd --network="${LIGHTNINGD_NETWORK}" "$@"` i.e. pass through directly.
2. With EXPOSE_TCP true, it started lightningd then waited, so it could start socat.
3. Now, I always have to start and wait, so I can run the scripts once it's started.

Thanks!

Reply:
> Seems we drop support for not exposing here? That's fine imho, since I don't think anybody wants to `docker exec` into the container in order to interact anyway, and exposing is likely done by the vast majority.

No, should still be optional.  And you can control by mounting the .lightning/ dir externally, without exposing our (UNSECURED!!!) RPC over TCP, or use commando.